docs/development/pigs/pig-008.rst

```diff
@@ -0,0 +1,449 @@
+        model=model,
+        likelihood="wstat"
+        )
+
+    fit = Fit(dataset)
+    fit.optimize()
+
+
+`SpectrumDataset`
+-----------------
+
+For spectral analysis we propose to introduce a `SpectrumDataset`:
+
+.. code ::
+
+    from gammapy.spectrum import SpectrumDataset
+
+    model = SpectralModel()
+    background_model = SpectralBackgroundModel.read()
+    edisp = EnergyDispersion.read()
+
+    counts = Spectrum.read()
+    exposure = Spectrum.read()
+    mask = Spectrum.read()
+
+    dataset = SpectrumDataset(
+        counts=counts,
+        exposure=exposure,
+        model=model,
+        background_model=background_model,
+        mask=mask,
+        edisp=edisp,
+        likelihood="cash"
+        )
+
+    dataset.npred()
+    dataset.likelihoop_per_bin()
+    dataset.likelihood(mask)
+
+The `SpectrumDatasets` with a parametric background model will be introduced first.
+The existing `SpectrumObservation` can be refactored later in a `SpectrumOnOffDataset`
+or `PHASpectrumDatasets`.
+
+
+`FluxPointDataset`
+------------------
+
+For modeling of flux points we propose to introduce a `FluxPointDataset`:
+
+.. code::
+
+    from gammapy.spectrum import FluxPointsDataset
+
+    flux_points = FluxPoints.read("flux_points.ecsv")
+    model = PowerLaw()
+    dataset = FluxPointsDataset(
+        flux_points=flux_points,
+        model=model,
+        likelihood="chi2"
+        )
+
+    fit = Fit(dataset)
+    fit.optimize()
+
+The `FluxPointDataset` should define at least the following methods:
+
+.. code::
+
+    dataset = FluxPointDataset()
+
+    dataset.likelihood_per_bin()
+    dataset.likelihood(mask)
+
+    # predicted flux form the spectral model
+    dataset.flux_pred()
+
+The `FluxPoint` class also supports the `likelihood` format, which has to be implemented
+in a special way in the `FluxPointDataset`. The `likelihood` format stores the
+likelihood of the flux point, depending on energy and amplitude. Given a predicted
+flux by the spectral model the likelihood can be directly interpolated. This could
+be supported by a specific option `FluxPointDataset(likelihood="template")` or similar.
+
+
+`Datasets`
+----------
+
+To combined multiple datasets in a joint-likelihood analysis we propose to introduce
+a `Datasets` class. It's repsonsibility is to add up the log-likelihood values of
+the individual datasets and join per dataset parameter lists. The `Datasets` class
+also represents the main interface to the `Fit` class.
+
+The following example shows how a joint-likelihood analysis of mutiple observations
+is supposed to work:
+
+.. code ::
+
+    from gammapy.utils.fitting import Datatsets
+
+    model = SourceModel.read()
+
+    map_datasets = []
+
+    for obs_id in [123, 124, ...]:
+        bkg_model = BackgroundModel.read("obs-{}/background.fits".format(obs_id))
+        dataset = MapDataset(model=model, background=bkg_model)
+        map_datasets.append(dataset)
+
+    datasets = Datasets(map_datasets)
```

Comment:
Add a `append` method to `Datasets`, and change to the pattern of creating an empty one via `datasets = Datasets()` and then appending to it directly?